app/main/views/suppliers.py

```diff
@@ -422,3 +377,103 @@ def register_interest_in_framework(supplier_id, framework_slug):
             return jsonify(message="Database Error: {0}".format(e)), 400
 
         return jsonify(frameworkInterest=interest_record.serialize()), 201
+
+
+@main.route('/suppliers/<supplier_id>/frameworks/<framework_slug>', methods=['PUT'])
+def register_framework_interest(supplier_id, framework_slug):
+    updater_json = validate_and_return_updater_request()
+
+    framework = Framework.query.filter(
+        Framework.slug == framework_slug
+    ).first_or_404()
+
+    supplier = Supplier.query.filter(
+        Supplier.supplier_id == supplier_id
+    ).first()
```

Comment:
Ah I've got it - because I cut & pasted from the (now deprecated) POST method above and didn't want to mess with the pattern.